programs/intent-transfer/src/intrachain/processor/send_tokens_with_fee.rs

```diff
@@ -0,0 +1,88 @@
+use crate::{
+    config::state::send_token_fee_config::{SendTokenFeeConfig, SEND_TOKEN_FEE_CONFIG_SEED},
+    intrachain::processor::send_tokens::*,
+};
+use anchor_lang::prelude::*;
+use anchor_spl::associated_token::get_associated_token_address;
+use anchor_spl::token::{transfer_checked, TransferChecked};
+use anchor_spl::{
+    associated_token::{self, AssociatedToken},
+    token::{Mint, Token, TokenAccount},
+};
+
+#[derive(Accounts)]
+pub struct SendTokensWithFee<'info> {
+    pub send_tokens: SendTokens<'info>,
+
+    #[account(mut, token::mint = fee_mint, token::authority = send_tokens.source.owner )]
+    pub fee_source: Account<'info, TokenAccount>,
+
+    #[account(init_if_needed, payer = send_tokens.sponsor, associated_token::mint = fee_mint, associated_token::authority = send_tokens.sponsor)]
```

Comment:
I don't think this should be `init_if_needed`; fees should never be paid to an uninitialized account so this seems like an error if it ever happens